Add SIWF SSO docs (#654)
ProjectLibertyLabs/gateway: docs/src/GettingStarted/BecomeProvider.md

# Become a Provider

A Provider is the on-chain representation of an application.
Any application that will interact with users must first register as a Provider.

> This guide will walk you through the steps to becoming a Provider on the Frequency Testnet. See [How to Become a Provider on Mainnet](#mainnet), if you are ready to move to production.

Reviewer: I'm not sure I agree with this description. Technically, a Provider represents an entity that wishes to run one **OR MORE** applications providing Frequency services to users.

Example:
Meta is a organization that runs the Instagram and Facebook applications
Meta could provision a single Provider and run both apps under it, or it could provision a separate Provider for each app.

Really, the decision of an organization of how to partition their applications amongst one or more registered Providers is a business decision that takes into account the economic model & how the organization wishes to leverage Capacity.

Author: Agreed, but not sure how to better state that for a getting started guide. Open to ideas.

Reviewer: ```suggestion
A Provider is a special kind of user account on Frequency, capable of executing certain operations on behalf of other users (delegators).
Any organization wishing to deploy an application that will act on behalf of users must first register as a Provider.
This guide will walk you through the steps to becoming a Provider on the Frequency Testnet. See [How to Become a Provider on Mainnet](#mainnet), if you are ready to move to production.
```

Author: Works for me. Committed.

- Visit the [Provider Dashboard](https://provider.frequency.xyz/)
- Select `Become a Provider`
- Select the `Testnet Paseo` network
- Connect the Application Account created earlier
- Select `Create an MSA` and approve the transaction popups
- Choose a public Provider name (e.g. "Cool Test App") and continue via `Create Provider`

## Step 4: Gain Capacity

Capacity is the ability to perform some transactions without token cost.
All interactions with the chain that an application does on behalf of a user can be done with Capacity.

In the [Provider Dashboard](https://provider.frequency.xyz/), login and select `Stake to Provider` and stake 100 XRQCY Tokens.

## Step 5: Done!

You are now registered as a Provider on Testnet and have Capacity to do things like support users with [Single Sign On](./SSO.md).

You can also use the [Provider Dashboard](https://provider.frequency.xyz/) to add additional Control Keys for safety.

## Ready to Become a Provider on Mainnet? { #mainnet }

Want to make the next step to becoming a Provider on Mainnet?

1. Securely generate a Frequency Mainnet Account
2. Backup your seed phrase for the account.

> 3. Acquire a small amount of FRQCY tokens.

Reviewer: Don't you need to acquire FRQCY tokens before you can generate a Frequency Mainnet Account?

Author: Account is just the key pair. One could say that the account doesn't exist until it has tokens, but I've usually used account to refer to public key (or keypair)

Reviewer: Hmm, could be confusing, as generating a control key pair in and of itself has no interaction with Frequency.

Reviewer: I understand the desire to speak in less blockchain-y, crtypo-ey terms, but perhaps we can assume that anyone seeking to build an app & integrate with our Gateway services is savvy enough to understand the concept of control keys?

Author: Account is the term that wallets use. Perhaps call it a Wallet Account? Not something we've done 

<img width="501" alt="image" src="https://github.com/user-attachments/assets/3fcd30d3-e875-4877-b254-ec34ba9469cf">

<img width="259" alt="image" src="https://github.com/user-attachments/assets/a3a537c6-ace6-4820-a78f-d4c6abca6516">

# Become a Provider

A Provider is the on-chain representation of an application.

> Those who which to interact with users as an application must register as a Provider.

Author: ```suggestion
Any application that will interact with users must first register as a Provider.
```

This guide will walk you through the steps to becoming a Provider on the Frequency Testnet. See [How to Become a Provider on Mainnet](#mainnet), if you are ready to move to production.

## Step 1: Generate Your Keys

> There are various wallets that can generate and secure Frequency compatible keys.

Author: ```suggestion
There are various wallets that can generate and secure Frequency compatible keys, including:
```

- [Polkadot Extension](https://polkadot.js.org/extension/)
- [Talisman](https://www.talisman.xyz)
- [See more](https://polkadot.com/get-started/wallets)

> The wallet onboarding process will guide you through the creation of the account.
> 
> This will be a Control Key for the registered Provider and will be required for many different interactions.

Author: ```suggestion
This onboarding process will guide you through the creation of an account and the creation of a Provider Control Key which will be required for many different transactions.
```
